sklearn/feature_extraction/tests/test_text.py

```diff
@@ -1199,14 +1199,16 @@ def build_preprocessor(self):
     'Estimator',
     [CountVectorizer,
      TfidfVectorizer,
-     pytest.param(HashingVectorizer, marks=fails_if_pypy)]
```

Comment:
For some reason, this doesn't seem to skip some of the tests with PyPy. I also saw tests with fixtures marked as  `@fails_if_pypy` still being executed (but only some), maybe something about PyPy and pytest. Anyway, I went with the less magic route,
```py
if IS_PYPY:
   pytest.xfail(...)
```
that works.

Reply:
xfail should still execute the text, yeah?

Reply:
No `pytest.xfail` will interrupt the test without running the rest (same as `pytest.skip`). The result will be always marked as xfail. To make it continue in this case, one would need to do something like
```py
def test_something(request):
    if IS_PYPY:
        request.applymarker(pytest.mark.xfail(run=False, reason='some reason'))
```
Then the result will be marked as either xfail or xpass depending on whether it fails or passes.

We could do it, but then we might as well do it in all other xfail occurences. I would rather do that in a separate PR.

I'm not sure how,
```py
pytest.param(HashingVectorizer, marks=fails_if_pypy)
```
was expected to work, but here that test was failing when trying to run HashingVectorizer which is explicitly not supported on PyPy.